prospero-common/src/main/java/org/wildfly/prospero/api/ArtifactChange.java

```diff
@@ -22,44 +22,25 @@
 
 import java.util.Optional;
 
-public class ArtifactChange {
-    private Artifact oldVersion;
-    private Artifact newVersion;
+public class ArtifactChange extends Diff {
 
     public ArtifactChange(Artifact oldVersion, Artifact newVersion) {
-        this.oldVersion = oldVersion;
-        this.newVersion = newVersion;
+        super(toGav(oldVersion == null?newVersion:oldVersion), oldVersion==null?null:oldVersion.getVersion(), newVersion==null?null:newVersion.getVersion());
```

Comment:
A detail - I like the approach in the ChannelChange, where the instances are created via static constructor methods. It would simplify this constructor code and as well make it clearer how to use the class.